spec/unit/actions/synchronize_user_filters_spec.rb

```diff
@@ -74,4 +54,59 @@
       end
     end
   end
+
+  context 'when using sync_one_project_based_on_custom_id mode with multiple_projects_column' do
+    let(:params) do
+      params = {
+        GDC_GD_CLIENT: client,
+        input_source: 'foo',
+        domain: 'bar',
+        filters_config: { labels: [] },
+        multiple_projects_column: 'id_column',
+        sync_mode: 'sync_one_project_based_on_custom_id'
+      }
+      GoodData::LCM2.convert_to_smart_hash(params)
+    end
+
+    context 'when params do not match client data in domain' do
+      before do
+        allow(project).to receive(:metadata).and_return({})
+        allow(project).to receive(:uri).and_return('project-uri')
+        allow(domain).to receive(:clients).and_return([])
+      end
+
+      it 'fails when unable to get filter value for selecting filters' do
+        expect { subject.class.call(params) }.to raise_exception
+      end
+    end
+
+    context 'when params match a client in the domain' do
+      let(:CSV) { double('CSV') }
+
+      before do
+        allow(project).to receive(:metadata).and_return(
+          'GOODOT_CUSTOM_PROJECT_ID' => 'project-123'
+        )
+        allow(project).to receive(:uri).and_return('project-uri')
+        allow(project).to receive(:add_data_permissions)
+        allow(domain).to receive(:clients).and_return([])
+        allow(data_source).to receive(:realize).and_return('filepath')
+      end
+
+      it 'adds filters matching the client' do
+        expect(File).to receive(:open)
+        csv_data = [
+          {
+            'id_column' => 'project-123'
+          },
+          {
+            'id_column' => 'another-project'
+          }
+        ]
+        expect(CSV).to receive(:foreach).and_yield(csv_data[0]).and_yield(csv_data[1])
+        expect(project).to receive(:add_data_permissions)
```

Comment:
Please check the params that go to `add_data_permissions`. E.g. `receive(:add_data_permissions).with(csv_data)`